instrumentation/runtime-metrics/javaagent/src/main/java/io/opentelemetry/instrumentation/javaagent/runtimemetrics/RuntimeMetricsInstaller.java

```diff
@@ -7,38 +7,43 @@
 
 import com.google.auto.service.AutoService;
 import io.opentelemetry.api.GlobalOpenTelemetry;
-import io.opentelemetry.instrumentation.api.config.Config;
 import io.opentelemetry.instrumentation.runtimemetrics.Classes;
 import io.opentelemetry.instrumentation.runtimemetrics.Cpu;
 import io.opentelemetry.instrumentation.runtimemetrics.GarbageCollector;
 import io.opentelemetry.instrumentation.runtimemetrics.MemoryPools;
 import io.opentelemetry.instrumentation.runtimemetrics.Threads;
 import io.opentelemetry.javaagent.extension.AgentListener;
-import io.opentelemetry.javaagent.tooling.config.AgentConfig;
 import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
-import java.util.Collections;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.util.Locale;
 
 /** An {@link AgentListener} that enables runtime metrics during agent startup. */
 @AutoService(AgentListener.class)
 public class RuntimeMetricsInstaller implements AgentListener {
 
-  private static final boolean DEFAULT_ENABLED =
-      Config.get().getBoolean("otel.instrumentation.common.default-enabled", true);
-
   @Override
-  public void afterAgent(Config config, AutoConfiguredOpenTelemetrySdk unused) {
-    if (new AgentConfig(config)
-        .isInstrumentationEnabled(Collections.singleton("runtime-metrics"), DEFAULT_ENABLED)) {
-
-      Classes.registerObservers(GlobalOpenTelemetry.get());
-      Cpu.registerObservers(GlobalOpenTelemetry.get());
-      MemoryPools.registerObservers(GlobalOpenTelemetry.get());
-      Threads.registerObservers(GlobalOpenTelemetry.get());
-
-      if (config.getBoolean(
-          "otel.instrumentation.runtime-metrics.experimental-metrics.enabled", false)) {
-        GarbageCollector.registerObservers(GlobalOpenTelemetry.get());
-      }
+  public void afterAgent(AutoConfiguredOpenTelemetrySdk autoConfiguredSdk) {
+    ConfigProperties config = autoConfiguredSdk.getConfig();
+
+    boolean defaultEnabled =
+        config.getBoolean(normalize("otel.instrumentation.common.default-enabled"), true);
+    if (!config.getBoolean(
+        normalize("otel.instrumentation.runtime-metrics.enabled"), defaultEnabled)) {
+      return;
     }
+
+    Classes.registerObservers(GlobalOpenTelemetry.get());
+    Cpu.registerObservers(GlobalOpenTelemetry.get());
+    MemoryPools.registerObservers(GlobalOpenTelemetry.get());
+    Threads.registerObservers(GlobalOpenTelemetry.get());
+
+    if (config.getBoolean(
+        normalize("otel.instrumentation.runtime-metrics.experimental-metrics.enabled"), false)) {
+      GarbageCollector.registerObservers(GlobalOpenTelemetry.get());
+    }
+  }
+
+  private static String normalize(String key) {
+    return key.toLowerCase(Locale.ROOT).replace('-', '.');
   }
```

Comment:
I found one more mismatch between `ConfigProperties` and `Config` -- in javaagent, we use the dash character in property names fairly often; turns out that SDK `ConfigProperties` does not support this kind of thing.
@jack-berg @anuraaga do you think we should bridge that gap in the SDK?

Reply:
I'm not familiar with the history of why `DefaultConfigProperties` replaces `-`, `_` with `.`. On the surface level it seems reasonable to perform the same normalization on keys when retrieving config values. Would defer to @jkwatson and @anuraaga though.

Reply:
SDK had just one at the time (and now two I think) properties with hyphens so it was just the easiest implementation given `ConfigProperties` was only generally used from autoconfigure implementation itself. Since then, it's exposed in many APIs and mysterious normalization is probably a bug :P

Reply:
> I'm not familiar with the history of why `DefaultConfigProperties` replaces `-`, `_` with `.`.

The `_` is replaced as a part of env var -> system prop name conversion. `-` was probably copied from the agent code (we replaced it with `.` to bridge the gap between the env vars and system properties).

> mysterious normalization is probably a bug :P

Is it? True, hyphens are not used in the SDK, but they're commonly used in the javaagent property names (I did some archeology work and found this comment explaining the reasoning behind it: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1747#issuecomment-733154848). Do we want `ConfigProperties` to behave similar to javaagent's `Config` with regard to this? I think that if we want to use `ConfigProperties` in javaagent SPIs to (for similarity with SDK SPIs) we should make `ConfigProperties` work with the kebab-case properties.

Reply:
> I did some archeology work

😄

> Do we want `ConfigProperties` to behave similar to javaagent's `Config` with regard to this?

I think so (pending approval from @open-telemetry/java-maintainers)

Reply:
> Is it? True, hyphens are not used in the SDK, but they're commonly used in the javaagent property names

Right I mean it's a bug in the SDK that it is normalizing away hyphens. Back when the autoconfigure's config was only used within autoconfigure, it wasn't a big deal that if a user specifies `my.custom-property` on the command line, it is only accessible as `.get("my.custom.property")`, but since it's exposed to users a lot now, it seems buggy. I can't remember what exactly was easier to go with the hacky normalization but we shouldn't do it.

Reply:
I've added an issue for that in the SDK repo: https://github.com/open-telemetry/opentelemetry-java/issues/4562
For now, I think we can merge this PR as it is and just leave a TODO note that this will be fixed separately.